solr/core/src/java/org/apache/solr/highlight/UnifiedSolrHighlighter.java

```diff
@@ -395,12 +395,12 @@ protected Locale parseLocale(String language, String country, String variant) {
             "language is required if country or variant is specified");
       } else if (country == null && variant != null) {
         throw new IllegalArgumentException("To specify variant, country is required");
-      } else if (country != null && variant != null) {
-        return new Locale(language, country, variant);
-      } else if (country != null) {
-        return new Locale(language, country);
       } else {
-        return new Locale(language);
+        return new Locale.Builder()
```

Comment:
this looks nicer!

Reply:
Earlier, the Locale constructors would throws NullPointerException if any of the param is Null. However, Locale Builder setters handle this case now and set the field to empty string if null is passed. Therefore, no need to explicitly check for null.